.github/workflows/ci-run-test.yml

```diff
@@ -230,7 +230,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip requests wheel urllib3 chardet build
       - name: Install poetry
-        run: pipx install poetry
+        run: pipx install poetry --python ${{ fromJson(inputs.version-set).python }}
```

Comment:
Poetry was always using the base python interpreter (3.10) instead of what we're testing.